server/ast/column_table_def.go

```diff
@@ -24,18 +24,18 @@ import (
 )
 
 // nodeColumnTableDef handles *tree.ColumnTableDef nodes.
-func nodeColumnTableDef(node *tree.ColumnTableDef) (_ *vitess.ColumnDefinition, err error) {
+func nodeColumnTableDef(node *tree.ColumnTableDef) (*vitess.ColumnDefinition, []*vitess.ConstraintDefinition, error) {
```

Comment:
Instead of returning `[]*vitess.ConstraintDefinition`, can we set the `vitess.ColumnDefinition::ColumnType::Constraint` property with the constraint definition and avoid the extra return param?

We can have check constraints defined on a column, or on a table, but it seems like this function is always dealing with check constraints defined on a column.

Reply:
Yes, the reason that we're returning the column check constraints here is that in vitess any column-declared check constraint is appended onto the table constraints and get handled as table constraints. https://github.com/dolthub/vitess/blob/main/go/vt/sqlparser/ast.go#L2674

In regards to the comment above, this is why the test for multiple check constraints on a single column from doltgres passes as expected although in dolt we don't support the syntax for it. I do think I might not be aware of edge cases failing from dropping the rest of the constraints except for the first one, so I agree we can error on that the case of multiple constraints on a single column.

Reply:
I think it's too hard to support multiple check constraint on a single column functionality in dolt, and so this won't be an issue here. But at the same time, I don't think it's of high priority. I decided to add this support to understand how check constraints work in dolt/gms better for implementing domain.

Reply:
Oh, interesting that `TableSpec.AddColumn` moves the checks up from the column to the table level! Thanks for sharing that link. 

I think a good approach could be to parse the `vitess.ColumnDefinition`, with the check included in the column type's `Constraint` field, then the code that calls `nodeColumnTableDef` could build the `TableSpec` by calling `AddColumn()` to add each column to the table. I like this approach, because it leaves the knowledge about how the constraint is structured in the `TableSpec` instance to the Vitess code, and doesn't require that the Doltgres code know about and be in sync with that – if it changes in the Vitess code, then we would be shielded from that in the doltgres code since we're going through the interface to build the TableSpec. I think it would still be fairly straightforward to support multiple checks on a column in this form, but I agree that we don't need to get distracted by that now. 

That said... I don't feel super strongly about this, but the above approach does feels a little cleaner and more encapsulated to me. 😉 Happy to continue discussing if you have concerns of course! 😄

Reply:
Ah using `AddColumn()` function makes sense! Thank you!